enterprise/frontend/src/metabase-enterprise/caching/components/utils.tsx

```diff
@@ -0,0 +1,9 @@
+import type { Dashboard } from "metabase-types/api";
+
+export const getDashboardId = (dashboard: Dashboard): number => {
+  if (typeof dashboard.id === "string") {
+    throw new Error("This dashboard has an invalid id");
+  }
+  const dashboardId: number = dashboard.id;
+  return dashboardId;
+};
```

Comment:
I've added some extra documentation. I don't yet see how to return null in a straightforward way here, since I need to pass the dashboardId to a hook, so I can't return null before calling that hook, and it seems overcomplicated to teach the hook how to handle a string id

Reply:
Actually I've now introduced a `CacheableDashboard` type which gets around the problem a bit more cleanly